cluster-essentials/clusterinit.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package clusteressentials has APIs for bootstrapping kubernetes cluster
+package clusteressentials
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/client-go/rest"
+
+	clusterutil "github.com/vmware-tanzu/tanzu-framework/cluster-essentials/util"
+)
+
+type ClusterEssentialsClient struct {
+	clusterEssentialBundlePath string
+	restConfig                 *rest.Config
+	pkgClient                  clusterutil.Client
+}
+
+func NewClusterEssentialsClient(clusterEssentialBundlePath string, restConfig *rest.Config, pkgClient clusterutil.Client) *ClusterEssentialsClient {
+	return &ClusterEssentialsClient{
+		clusterEssentialBundlePath: clusterEssentialBundlePath,
+		restConfig:                 restConfig,
+		pkgClient:                  pkgClient,
+	}
+}
+
+const tmpDir = "tmp"
+
+func (c *ClusterEssentialsClient) InitClusterEssentials() error {
+	err := c.pkgClient.DownloadClusterEssentialBundle(c.clusterEssentialBundlePath)
+	if err != nil {
+		return fmt.Errorf("unable to download cluster essential manifest: %w", err)
+	}
+
+	packages, err := os.ReadDir(tmpDir)
+	if err != nil {
+		return err
+	}
+	var generatedManifestBytes []byte
+	for _, item := range packages {
+		packageDir := filepath.Join(tmpDir, item.Name())
+		generatedManifestByte, err := c.pkgClient.CarvelPackageProcessor(packageDir)
+		generatedManifestBytes = append(generatedManifestBytes, generatedManifestByte...)
+		if err != nil {
+			return fmt.Errorf("unable to process carvel package %s: %w", item.Name(), err)
+		}
+	}
+
+	err = c.pkgClient.ApplyResourcesFromManifest(generatedManifestBytes, c.restConfig)
+	if err != nil {
+		return fmt.Errorf("unable to install : %w", err)
+	}
```

Comment:
This does not look like a blocking API. This method needs to block i.e. not return until everything is installed within a timeout or error out. Otherwise, consuming code wouldn't know things are successful or not.

Ideally, we need to have 2 APIs like:

```go
// Install with default timeout
func Install() error {
    // 1. Apply manifests
    // 2. Pick a default timeout
    // 3. Wait for results
}

// Install with option to provide a timeout
func InstallWithTimeout(timeout time.Duration) error {
    // 1. Apply manifests
    // 2. Wait for results
}
```